doc/tests/verification_plan.rst

```diff
@@ -0,0 +1,138 @@
+########################
+
+This document provides a general introduction of the Zephyr Verification
+Plan. Zephyr provides test suite for various functionalities offered in
+the project like:
+
+**Kernel services and capabilities**
+    Services offered by kernel like memory management, scheduling,
+    interrupt services, timers and data passing mechanisms are verified with API
+    level testing and functionality testing.
+
+**Networking services**
+    Various protocols including HTTP, COAP, MQTT, ARP, 6lo, TCP, UDP
+    and other protocols are verified.
+
+**Sensor subsystem support**
+    Zephyr sensor driver is validated to measure various physical parameters
+    such as temperature, pressure, ambience, humidity, light intensity,
+    proximity and other real time sensory information.
+
+**File system**
+    Newtron Flash Filesystem (NFFS) and FATFS support, FCB (Flash Circular Buffer)
+    for memory constrained applications, and file system enhancements for logging
+    and configuration are verified.
+
+**Device Driver support**
+    Device drivers such as GPIO, I2C, I2S, SPI, RTC, UART, AIO and various other
+    driver functionalities are verified.
+
+**Power management services**
+    Power management services such as tickless idle and an advanced idling
+    infrastructure are also verified.
+
+**POSIX compliance**
+    APIs introduced to make Zephyr POSIX complaint are verified through API level testing.
+
+Verification Strategy and Approach
+##################################
+
+Zephyr test suite contains various test types such as
+
+#. Component level test
+
+   Test against single Zephyr module.
+
+#. API level test
+
+   Test correctness of Zephyr APIs made available for application developer.
+
+#. Stress test
+
+   Thorough and intense tests to ensure robustness of Zephyr components.
+
+#. Security test
+
+   Intended to reveal flaws in the security mechanisms such memory protection
+   implementation and stack corruption check.
+
+#. Performance test
+
+   Testing done against key performance indicators to ensure effectiveness
+   of Zephyr.
+
+#. Integration test
+
+   Individual components are combined and verified as a group.
+
+Zephyr Test Framework - Ztest
+#############################
+
+The Zephyr Test Framework (Ztest) provides a simple testing framework intended to be used during development.
+It provides basic assertion macros and a generic test structure. The framework can be used in two ways,
+either as a generic framework for integration testing, or for unit testing specific modules.
+
+Each test name has to be unique, prefixed with test\_ and documented using doxygen.
+
+Debug and result analysis using prints are done using TC_PRINT function.
+
+Detailed explanation about the APIs and its usage in test application is given in this document.
+http://docs.zephyrproject.org/subsystems/test/ztest.html
+
+Test Methodology
+################
+
+**Test Setup**
+
+Zephyr verification setup contains
+
+#. Test Client
```

Comment:
As before, use a dictionary list instead of a numbered list:

    Test Client
       Test client triggers the test, the test can be triggered per
       component or pre test suite.

    Device under test
       Device under test could be various boards belonging to
       different architectures supported by Zephyr.

etc.